packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_route.ts

```diff
@@ -148,7 +149,8 @@ export class CoreVersionedRoute implements VersionedRoute {
 
     const response = await handler.fn(ctx, mutableCoreKibanaRequest, res);
 
-    if (this.router.validateResponses && validation?.response?.[response.status]) {
+    const core = await (ctx as RequestHandlerContext).core;
+    if (core.env.mode.dev && validation?.response?.[response.status]) {
```

Comment:
All changes made enable us to do this check using context at the request handler level.